Sources/Sentry/SentryCrashWrapper.m

```diff
@@ -183,16 +184,18 @@ - (void)enrichScope:(SentryScope *)scope
 
 // The UIWindowScene is unavailable on visionOS
 #if SENTRY_TARGET_REPLAY_SUPPORTED
-
-    NSArray<UIWindow *> *appWindows = SentryDependencyContainer.sharedInstance.application.windows;
-    if ([appWindows count] > 0) {
-        UIScreen *appScreen = appWindows.firstObject.screen;
-        if (appScreen != nil) {
-            [deviceData setValue:@(appScreen.bounds.size.height) forKey:@"screen_height_pixels"];
-            [deviceData setValue:@(appScreen.bounds.size.width) forKey:@"screen_width_pixels"];
+    if (options.enableAutoSessionTracking) {
+        NSArray<UIWindow *> *appWindows
+            = SentryDependencyContainer.sharedInstance.application.windows;
+        if ([appWindows count] > 0) {
+            UIScreen *appScreen = appWindows.firstObject.screen;
+            if (appScreen != nil) {
+                [deviceData setValue:@(appScreen.bounds.size.height)
+                              forKey:@"screen_height_pixels"];
+                [deviceData setValue:@(appScreen.bounds.size.width) forKey:@"screen_width_pixels"];
+            }
```

Comment:
`h`: this option is not relevant here, we should ensure the `enrichScope` is called from the main thread instead.

Reply:
Yes, indeed. The `enrichScope` is called on the main thread when starting the SDK. We could also cache `screen_height_pixels` and `screen_width_pixels`.